tiledb/sm/array/array.cc

```diff
@@ -200,12 +201,17 @@ Status Array::open(
     EncryptionType encryption_type,
     const void* encryption_key,
     uint32_t key_length) {
-  std::unique_lock<std::mutex> lck(mtx_);
-
+  // Checks
   if (is_open_)
     return LOG_STATUS(
         Status_ArrayError("Cannot open array; Array already open"));
+  bool is_array = false;
+  RETURN_NOT_OK(storage_manager_->is_array(array_uri_, &is_array));
```

Comment:
This check will break REST arrays. `is_array` does not support checking a remote array, it simply does some checking via VFS. Either we need to adjust the `is_array` function or more likely move this into the else blocks of the `if (remote_) {`

Reply:
Hmm I wondered about that, thanks for pointing it out. I will adjust.

Reply:
To further my thoughts for anyone who stumbles upon this. We could have `is_array` check for remote arrays but this would introduce additional request and latency. The checking for the array is implicit in the open sequence when array schema and other details are fetched for the remote array. Since we have an implicit check we can avoid an explicit one and not introduce the additional (several hundred MS realistically) latency to the opening of the array.

Reply:
Would just `if (!remote_ && !is_array)` fix the problem?

Reply:
I mean I would make sure that `is_array` is called only if `!remote_`.